src/cublas/laset.cu

```diff
@@ -0,0 +1,134 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include "dlaf/cublas/laset.h"
+#include "dlaf/cuda/assert.cu.h"
+#include "dlaf/util_cuda.h"
+#include "dlaf/util_math.h"
+
+namespace dlaf::gpublas {
+namespace kernels {
+
+using namespace dlaf::util::cuda_operators;
+
+struct LasetParams {
+  static constexpr unsigned kernel_tile_size_rows = 64;
+  static constexpr unsigned kernel_tile_size_cols = 16;
+};
+
+__device__ inline constexpr bool all(unsigned i, unsigned j) noexcept {
+  return true;
+}
+
+__device__ inline constexpr bool lower(unsigned i, unsigned j) noexcept {
+  return i >= j;
+}
+
+__device__ inline constexpr bool upper(unsigned i, unsigned j) noexcept {
+  return i <= j;
+}
+
+template <class T>
+__device__ void setAll(const unsigned m, const unsigned n, const T alpha, T* a, const unsigned lda) {
+  constexpr unsigned kernel_tile_size_rows = LasetParams::kernel_tile_size_rows;
+  constexpr unsigned kernel_tile_size_cols = LasetParams::kernel_tile_size_cols;
```

Comment:
Since we are in `__device__` code, why not using [`blockDim`](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#built-in-variables)?

Reply:
In general I prefer to use constexpr values as they allows the compiler to optimize better.